src/py21cmfast/_memory.py

```diff
@@ -0,0 +1,706 @@
+    if flag_options.USE_TS_FLUCT:
+        stored_memory += 2.0 * memory_st["python"]
+
+        # We also have initialied C memory that is retained until the end of the calculation
+        stored_memory += memory_st["c_init"]
+
+    # Add the two brightness_temperature boxes
+    stored_memory += 2.0 * memory_bt["python"]
+
+    # Now we have an estimate for the data retained in memory, now we just need to check
+    # the peak memory usage (which includes the additional C memory that is allocated
+    # and then freed on a single redshift call)
+    # First check perturb field
+    current_memory = stored_memory + memory_pf["c"]  # Add the temporary C memory
+
+    # Check if our peak memory usage has been exceeded
+    peak_memory = peak_memory if peak_memory > current_memory else current_memory
+
+    # Check spin temperature
+    if flag_options.USE_TS_FLUCT:
+        current_memory = stored_memory + memory_st["c_per_z"]
+
+        # Check if our peak memory usage has been exceeded
+        peak_memory = peak_memory if peak_memory > current_memory else current_memory
+
+    # Check ionized box
+    current_memory = stored_memory + memory_ib["c"]
+
+    # Check if our peak memory usage has been exceeded
+    peak_memory = peak_memory if peak_memory > current_memory else current_memory
+
+    # Check brightness temperature
+    current_memory = stored_memory + memory_bt["c"]
+
+    # Check if our peak memory usage has been exceeded
+    peak_memory = peak_memory if peak_memory > current_memory else current_memory
+
+    memory_data.update({"peak_memory": peak_memory})
+
+    format_output(
+        memory_data=memory_data,
+        user_params=user_params,
+        astro_params=astro_params,
+        flag_options=flag_options,
+    )
+
+    return memory_data
+
+
+def mem_initial_conditions(
+    *,
+    user_params=None,
+):
+    """A function to estimate total memory usage of an initial_conditions call."""
+    """Memory usage of Python InitialConditions class."""
+    """All declared HII_DIM boxes"""
+    # lowres_density, lowres_vx, lowres_vy, lowres_vz, lowres_vcb
+    # lowres_vx_2LPT, lowres_vy_2LPT, lowres_vz_2LPT
+    num_py_boxes_HII_DIM = 8.0
+
+    """All declared DIM boxes"""
+    # hires_density, hires_vx, hires_vy, hires_vz
+    # hires_vx_2LPT, hires_vy_2LPT, hires_vz_2LPT
+    num_py_boxes_DIM = 7.0
+
+    size_py = num_py_boxes_DIM * (user_params.DIM) ** 3
+    size_py += num_py_boxes_HII_DIM * (user_params.HII_DIM) ** 3
+
+    # These are all float arrays
+    size_py = (np.float32(1.0).nbytes) * size_py
+
+    """Memory usage within GenerateICs"""
+    kspace_num_pixels = (float(user_params.DIM) / 2.0 + 1.0) * (user_params.DIM) ** 2
+
+    """All declared DIM boxes"""
+    # HIRES_box, HIRES_box_saved
+    num_c_boxes = 2
+
+    """All declared 2LPT boxes (DIM)"""
+    # phi_1 (6 components)
+    if global_params.SECOND_ORDER_LPT_CORRECTIONS:
+        num_c_boxes += 6
```

Comment:
just a reminder to change this to 1

Reply:
Yep, this will be changed as the various PRs are merged